### PR TITLE
Stackelberg compiler and verifier

### LINKED ISSUE #10: Make `Meta Action Verifier`
Using some of the other toolchain and the stackleberg planner

### PR DESCRIPTION
**Pull request by kris701**
closes #10

### CODE REVIEW

File: StacklebergCompiler/ReservedNames.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace StacklebergCompiler
+{
+    public static class ReservedNames
+    {
+        // Prefixes
+        public static readonly string LeaderActionPrefix = "fix_";
+        public static readonly string FollowerActionPrefix = "attack_";
```
Comment (jamadaha) [inline StacklebergCompiler/ReservedNames.cs:13]: Is there a reason to fix and attack? Wasn't legacy from before they figured out better terminology

**Review by jamadaha — COMMENTED** [review]
How do I run this? I get errors when trying to run the Verifier
```
dotnet run --project StackelbergVerifier/ --domain "/home/jamadaha/GitH
ub/P9/StacklebergCompiler/TestData/gripper-domain.pddl" --problem "/home/jamadaha/GitHub/P9/StacklebergCompiler
/TestData/gripper-prob1.pddl"
Verifying paths...
Unhandled exception. System.IO.FileNotFoundException: Stackelberg planner file not found: /home/jamadaha/GitHub
/P9/fast-downward.py
   at StacklebergVerifier.Program.RunStacklebergVerifier(StackelbergVerifierOptions opts) in /home/jamadaha/Git
Hub/P9/StackelbergVerifier/Program.cs:line 35
   at CommandLine.ParserResultExtensions.WithParsed[T](ParserResult`1 result, Action`1 action)
   at StacklebergVerifier.Program.Main(String[] args) in /home/jamadaha/GitHub/P9/StackelbergVerifier/Program.c
s:line 15
```
This is after running setup.

**Comment by kris701** [comment]
> How do I run this? I get errors when trying to run the Verifier
> 
> ```
> dotnet run --project StackelbergVerifier/ --domain "/home/jamadaha/GitH
> ub/P9/StacklebergCompiler/TestData/gripper-domain.pddl" --problem "/home/jamadaha/GitHub/P9/StacklebergCompiler
> /TestData/gripper-prob1.pddl"
> Verifying paths...
> Unhandled exception. System.IO.FileNotFoundException: Stackelberg planner file not found: /home/jamadaha/GitHub
> /P9/fast-downward.py
>    at StacklebergVerifier.Program.RunStacklebergVerifier(StackelbergVerifierOptions opts) in /home/jamadaha/Git
> Hub/P9/StackelbergVerifier/Program.cs:line 35
>    at CommandLine.ParserResultExtensions.WithParsed[T](ParserResult`1 result, Action`1 action)
>    at StacklebergVerifier.Program.Main(String[] args) in /home/jamadaha/GitHub/P9/StackelbergVerifier/Program.c
> s:line 15
> ```
> 
> This is after running setup.

You arent providing the path to the stackelberg planner
There is an argument for it (you can see it with --help), where you can give the path to the planner

File: StacklebergCompiler/ReservedNames.cs (hunk 1 above)
Comment (kris701) [reply StacklebergCompiler/ReservedNames.cs:13]: These names are the names the Stackelberg Planner uses
We would need to modify the Planner to be able to use other names, and i would rather not do that

**Comment by kris701** [comment]
Also, do note that the `StackelbergVerifyer` is for verifying a *compiled* PDDL domain
The *StackelbergCompiler* is for making the actual compiled domains

**Review by jamadaha — APPROVED** [review]

**Pull request merged**